src/components/PersistentBar/PersistentBar.tsx

```diff
@@ -0,0 +1,192 @@
+import React, { FC, ReactNode, Ref } from 'react';
+import { Icon, IconName } from '../Icon';
+import { PersistentBarsProps, PersistentBarType } from './PersistentBar.types';
+import { mergeClasses } from '../../shared/utilities';
+import {
+    PrimaryButton,
+    DefaultButton,
+    SecondaryButton,
+    ButtonType,
+} from '../Button';
+import { Pagination, PaginationLayoutOptions } from '../Pagination';
+
+import styles from './persistentBar.module.scss';
+
+export const PersistentBar: FC<PersistentBarsProps> = React.forwardRef(
+    (
+        {
+            buttonMenuProps,
+            content,
+            icon,
+            type = PersistentBarType.bottomBarWithText,
+            closable,
+            onClose,
+            style,
+            classNames,
+            closeIcon = IconName.mdiClose,
+            closeButtonProps,
+            actionButtonOneProps,
+            actionButtonTwoProps,
+            actionButtonThreeProps,
+            role = 'presentation',
+            title,
+            paginationArgs,
+            paginationTotal,
+            ...rest
+        },
+        ref: Ref<HTMLDivElement>
+    ) => {
+        const persistentBarClasses: string = mergeClasses([
+            styles.persistentBar,
+            classNames,
+            {
+                [styles.bottomWithText]:
+                    type === PersistentBarType.bottomBarWithText,
+            },
+            {
+                [styles.bottomSecondaryButtons]:
+                    type === PersistentBarType.bottomBarSecondaryButtons,
+            },
+            {
+                [styles.bottomLeftAlign]:
+                    type === PersistentBarType.bottomBarButtonsOnLeft,
+            },
+            {
+                [styles.topButtonMenu]:
+                    type === PersistentBarType.topBarButtons,
+            },
+            { [styles.topWithText]: type === PersistentBarType.topBarWithText },
+            {
+                [styles.topPagination]:
+                    type === PersistentBarType.topBarPagination,
+            },
+        ]);
+
+        const getIconName = (): IconName => {
+            if (icon) {
+                return icon;
+            }
+            switch (type) {
+                case PersistentBarType.topBarWithText:
+                    return IconName.mdiArrowLeft;
+            }
+        };
+
+        const getTexts = (): ReactNode => {
```

Comment:
Not all codepaths return a value. You may either return null after the switch or add a default of no-op to get around this error.